hcatalog/server-extensions/src/main/java/org/apache/hive/hcatalog/listener/DbNotificationListener.java

```diff
@@ -858,19 +859,65 @@ private void addNotificationLog(NotificationEvent event, ListenerEvent listenerE
       long nextNLId = getNextNLId(stmt, sqlGenerator,
               "org.apache.hadoop.hive.metastore.model.MNotificationLog");
 
-      String insertVal = "(" + nextNLId + "," + nextEventId + "," + now() + ", ?, ?," +
-              quoteString(" ") + ",?, ?)";
+      String insertVal;
+      String columns;
+      List<String> params = new ArrayList<String>();
+
+      // Construct the values string, parameters and column string step by step simultaneously so
+      // that the positions of columns and of their corresponding values do not go out of sync.
+
+      // Notification log id
+      columns = "\"NL_ID\"";
+      insertVal = "" + nextNLId;
+
+      // Event id
+      columns = columns + ", \"EVENT_ID\"";
+      insertVal = insertVal + "," + nextEventId;
+
+      // Event time
+      columns = columns + ", \"EVENT_TIME\"";
+      insertVal = insertVal + "," + now();
+
+      // Event type
+      columns = columns + ", \"EVENT_TYPE\"";
+      insertVal = insertVal + ", ?";
+      params.add(event.getEventType());
+
+      // Message
+      columns = columns + ", \"MESSAGE\"";
+      insertVal = insertVal + ", ?";
+      params.add(event.getMessage());
+
+      // Message format
+      columns = columns + ", \"MESSAGE_FORMAT\"";
+      insertVal = insertVal + ", ?";
+      params.add(event.getMessageFormat());
+
+      // Database name, optional
+      if (event.getDbName() != null) {
+        columns = columns + ", \"DB_NAME\"";
+        insertVal = insertVal + ", ?";
+        params.add(event.getDbName());
```

Comment:
Can insert the Db, table and catalog names in lower case to make sure the query on it later works.

Reply:
By following the code where the NotificationEvent is created and db, cat and table name are set, I can confirm that they are always lower case. Using toLowercase() again might waste some CPU cycles (neglegible, but non-zero). In order to make sure that no future change here causes an upper case name to be added to the matastore, may be we could add an assert to make sure that the names are all lower case. Would that work?

Reply:
Assert sounds good to me.

Reply:
I added the assertion. With the help of this assertion I found that the call to addNotificationLog() from onAllocWriteId() was passing database name without converting it to the lower case, whereas rest of that function was using a database name converted to lower case. FIxed that case as well.